zyte_parsers/utils.py

```diff
@@ -55,19 +56,22 @@ def strip_urljoin(base_url: Optional[str], url: Optional[str]) -> str:
     return urljoin(base_url or "", url or "")
 
 
-def extract_link(a_node: SelectorOrElement, base_url: str) -> Optional[str]:
+def add_https_to_url(url: str) -> str:
+    if url.startswith(("http://", "https://")):
+        return url
+
+    parsed_url = urlparse(url)
+    if not parsed_url.scheme and parsed_url.netloc:
+        parsed_url = parsed_url._replace(scheme="https")
+
+    return str(urlunparse(parsed_url))
```

Comment:
Hm, why do we need this function? If a URL is written without http or https, it won't work anyways in the browser, right (unless it's a relative URL, which uses different rules - adding http/https is wrong for them)? 

If so, why are we adding the missing schema? My concern is that it'd change some invalid URLs to URLs which look valid, but which are not.

Reply:
Hm, or is it the `//` example, i.e. the relative schema ("use whatever protocol the page is using")? In this case, adding https is not 100% precise, as it should be the protocol of the web page itself.

Reply:
Yes, the idea is to handle such cases like //example.com when `base_url` here https://github.com/zytedata/zyte-parsers/pull/19/files#diff-0db3ea2158d725e2de6295c907fbe44928efc0c94057ef6ca85dbbee4c733819R83 is not provided.